### PR TITLE
Update all dependencies + fix clippy warnings

### PR DESCRIPTION
**Pull request by sigmaSd**

### CODE REVIEW

File: src/colorspace.rs [hunk 1]
```diff
@@ -3,6 +3,8 @@ use crate::Color;
 
 pub trait ColorSpace {
     fn from_color(c: &Color) -> Self;
+    // NOTE: Clippy's suggestion might be correct here
+    #[allow(clippy::wrong_self_convention)]
```
Comment (sharkdp) [inline src/colorspace.rs:7]: so what happens if we change it to `self`?

File: Cargo.toml [hunk 2]
```diff
@@ -14,34 +14,34 @@ exclude = ["doc/pastel.gif"]
...
-lazy_static = "1.3.0"
-rand = "0.7"
-atty = "0.2"
-output_vt100 = "0.1"
-nom = "5"
+lazy_static = "1.4.0"
+rand = "0.8.4"
+atty = "0.2.14"
+output_vt100 = "0.1.2"
+nom = "7.0.0"
```
Comment (sharkdp) [inline Cargo.toml:21]: Why did you add minor versions here? This makes maintenance hard, IMO. The lock file already stores the compatible minor versions (which can be used with `cargo … --locked`)

**Comment by sharkdp** [comment]
Thank you very much. This looks great :+1:

File: src/colorspace.rs (hunk 1 above)
Comment (sigmaSd) [reply src/colorspace.rs]: Seems to not require much work, I didn't even try it xD

File: Cargo.toml (hunk 2 above)
Comment (sigmaSd) [reply Cargo.toml:21]: I just run cargo upgrade from cargo-edit, I can revert that

Comment (sharkdp) [reply Cargo.toml:21]: Yes please.

Comment (sigmaSd) [reply Cargo.toml]: I updated as you suggested, is that correct now?

**Review by sharkdp — APPROVED** [review]
Thank you!
